ambari-server/src/main/java/org/apache/ambari/server/api/services/AlertTargetService.java

```diff
@@ -33,54 +33,165 @@
 import javax.ws.rs.core.UriInfo;
 
 import org.apache.ambari.annotations.ApiIgnore;
+import org.apache.ambari.server.api.resources.AlertTargetResourceDefinition;
 import org.apache.ambari.server.api.resources.ResourceInstance;
+import org.apache.ambari.server.controller.AlertTargetSwagger;
 import org.apache.ambari.server.controller.spi.Resource;
 import org.apache.ambari.server.state.alert.AlertTarget;
+import org.apache.http.HttpStatus;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+
 
 /**
  * The {@link AlertTargetService} handles CRUD operation requests for alert
  * targets.
  */
 @Path("/alert_targets/")
+@Api(value = "/alerts", description = "Endpoint for alert specific operations")
 public class AlertTargetService extends BaseService {
 
-  @GET @ApiIgnore // until documented
+  public static final String ALERT_TARGET_REQUEST_TYPE = "org.apache.ambari.server.controller.AlertTargetSwagger";
+
+  /**
+   * Handles GET /alert_targets
+   * Get all alert targets.
+   *
+   * @param headers http headers
+   * @param ui      uri info
+   * @return alert target resource representation
+   */
+  @GET @ApiIgnore // documented below
```

Comment:
I think `/alert_targets` and `/alert_targets/{targetId}` should be documented separately, since they have a few differences.